src/App.jsx

```diff
@@ -98,34 +120,32 @@ function App() {
       body: JSON.stringify(payload),
     };
     try {
-      const resp = await fetch(url, options);
+      const resp = await fetch(
+        encodeUrl({ sortField, sortDirection, queryString }),
+        options
+      );
       if (!resp.ok) {
         throw new Error(resp.status);
       }
       const { records } = await resp.json();
-      const updatedTodo = { id: records[0]["id"], ...records.fields };
+      const updatedTodo = { id: records[0]["id"], ...records[0].fields };
```

Comment:
This additional `updateTodo` is just adding unnecessary overhead. You are applying an optimistic approach to your UI update. This can be removed.